services/migrations/github_exported_data.go

```diff
@@ -0,0 +1,1335 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
+	base "code.gitea.io/gitea/modules/migration"
+
+	"github.com/hashicorp/go-version"
+)
+
+/*
+	{"version":"1.0.1","github_sha":"8de0984858fd99a8dcd2d756cf0f128b9161e3b5"}
+*/
+type githubSchema struct {
+	Version   string
+	GithubSha string `json:"github_sha"`
+}
+
+/*
+{
+    "type": "user",
+    "url": "https://github.com/sunvim",
+    "avatar_url": "https://avatars.githubusercontent.com/u/859692?v=4",
+    "login": "sunvim",
+    "name": "mobus",
+    "bio": "code happy ",
+    "company": "Ankr",
+    "website": null,
+    "location": "Shanghai",
+    "emails": [
+      {
+        "address": "sv0220@163.com",
+        "primary": true,
+        "verified": true
+      }
+    ],
+    "billing_plan": null,
+    "created_at": "2011-06-19T11:25:35Z"
+  },
+*/
+type githubUser struct {
+	URL       string
+	AvatarURL string `json:"avatar_url"`
+	Login     string
+	Name      string
+	Bio       string
+	Company   string
+	Website   string
+	Location  string
+	Emails    []struct {
+		Address  string
+		Primary  bool
+		Verified bool
+	}
+	CreatedAt time.Time `json:"created_at"`
+}
+
+func getID(s string) int64 {
```

Comment:
It's better be `getXxxID`, and maybe the newly written code had better to put in a separate package.

Reply:
This function has been used to get user, issue, attachment and etc. ids. I have no idea for a better name. And since it's only used by github exported data. I think currently it could stay here.

Reply:
`parseGitHubUserIDFromExportedURL`

Reply:
It's not only for User

Reply:
There could be a concept. Maybe `parseGitHubResID`. 

Anyway, just my opinion, either is fine.